### PR TITLE
Wsc 013 publish temprature scales

### LINKED ISSUE #29: refactor design for diffent scales of sensor data
### Provide temeprature data in 
* kelvin
* celcius
* fahrenheit

#### Use CRTP pattern instead of runtime polymorphism.

### PR DESCRIPTION
**Pull request by ggulgulia**

### CODE REVIEW

File: sensor_interface/headers/TemperatureScale.hpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    };
+
+    class Kelvin : public TemperatureScale{
+        public:
+            Kelvin() = default;
+            Kelvin(const Kelvin&) = default;
+             virtual ~Kelvin() = default;
+            double getValue()const noexcept override final;
+            std::ostream&  format(std::ostream& os) const override final;
+
```
Comment (Anokhi1994) [inline sensor_interface/headers/TemperatureScale.hpp]: remove empty line

File: sensor_interface/headers/TemperatureSensor.hpp [hunk 2]
```diff
@@ -1,31 +1,41 @@
...
+    class TemperatureSensor : public WeatherSensorInterface {
+    public:
+        const TempSensorPowerMap state_map{{"0", state::off}, {"1", state::on}, {"-19", state::ground_disconnected}};
+
+        TemperatureSensor():temperature_{std::make_unique<Celcius>()}
+        {   }
+
+        TemperatureSensor(std::unique_ptr<TemperatureScale>& temp):temperature_{std::move(temp)}
+        {   }
+
```
Comment (Anokhi1994) [inline sensor_interface/headers/TemperatureSensor.hpp]: * remove empty lines
*move implementation to .cpp file

File: sensor_interface/headers/TemperatureSensor.hpp [hunk 3]
```diff
@@ -1,31 +1,41 @@
...
+        {   }
+        
+        //avoid creating copies of sensor
+        TemperatureSensor(const TemperatureSensor&) = delete;
+        TemperatureSensor& operator=(const TemperatureSensor&) = delete;
+        virtual ~TemperatureSensor() = default;
+        state is_on() const noexcept override;
+        void update_temperature_sensor_power_status() noexcept(false);
+        float check_temperature() const noexcept(false);
+
```
Comment (Anokhi1994) [inline sensor_interface/headers/TemperatureSensor.hpp]: remove empty lines

File: sensor_interface/headers/TemperatureSensor.hpp [hunk 4]
```diff
@@ -1,31 +1,41 @@
...
+    private:
+        state state_;
+        std::unique_ptr<TemperatureScale> temperature_{nullptr};
     };
 
+
```
Comment (Anokhi1994) [inline sensor_interface/headers/TemperatureSensor.hpp]: remove empty line

File: sensor_interface/test/unit_tests/read_temperature_test.cpp [hunk 5]
```diff
@@ -1,34 +1,36 @@
...
 }
 
-
-int main(int argc, char** argv){
+int main(int argc, char** argv)
+{
     testing::InitGoogleTest(&argc, argv);
-  return RUN_ALL_TESTS();
+    return RUN_ALL_TESTS();
 }
```
Comment (Anokhi1994) [inline sensor_interface/test/unit_tests/read_temperature_test.cpp:36]: return to old style

**Review by Anokhi1994 — CHANGES_REQUESTED** [review]
please fix the comments

File: sensor_interface/headers/TemperatureSensor.hpp (hunk 2 above)
Comment (ggulgulia) [reply sensor_interface/headers/TemperatureSensor.hpp]: done

File: sensor_interface/headers/TemperatureSensor.hpp (hunk 3 above)
Comment (ggulgulia) [reply sensor_interface/headers/TemperatureSensor.hpp]: done

File: sensor_interface/headers/TemperatureScale.hpp (hunk 1 above)
Comment (ggulgulia) [reply sensor_interface/headers/TemperatureScale.hpp]: done

**Review by ggulgulia — COMMENTED** [review]
done reviewing

**Review by Anokhi1994 — APPROVED** [review]

**Pull request merged**